wrangler-core/src/main/java/io/cdap/wrangler/utils/ArithmeticOperations.java

```diff
@@ -24,380 +24,462 @@
    */
-  public static Double divider(Double x, Double y) {
-    if (x == null || y == null || y == 0) {
+  public static Number minus(Number x, Number y) throws DirectiveExecutionException {
+    NumberType outputType = getOutputType(x, y);
+    if (outputType == null) {
       return null;
     }
-    return x % y;
-  }
-
-  /**
-   * Arithmetic operation - Divide a column by another and return the remainder.
-   */
-  public static Float divider(Float x, Float y) {
-    if (x == null || y == null || y == 0) {
-      return null;
+    switch (outputType) {
+      case SHORT:
+        return (short) (x.shortValue() - y.shortValue());
+      case INTEGER:
+        return x.intValue() - y.intValue();
+      case LONG:
+        return x.longValue() - y.longValue();
+      case FLOAT:
+        return x.floatValue() - y.floatValue();
+      case DOUBLE:
+        return x.doubleValue() - y.doubleValue();
+      default:
+        BigDecimal bdX = new BigDecimal(x.toString());
+        BigDecimal bdY = new BigDecimal(y.toString());
+        return bdX.subtract(bdY).stripTrailingZeros();
     }
-    return x % y;
   }
 
   /**
-   * Arithmetic operation - Divide a column by another and return the remainder.
+   * Arithmetic operation - multiply any number of columns, of any types.
+   * Output type is most general of input types.
+   * Returns null if any input value is null.
    */
-  public static BigDecimal divider(BigDecimal x, BigDecimal y) {
-    if (x == null || y == null || y.equals(BigDecimal.valueOf(0))) {
+  public static Number multiply(Number... nums) throws DirectiveExecutionException {
+    NumberType outputType = getOutputType(nums);
+    if (outputType == null) {
       return null;
     }
-    return x.remainder(y);
-  }
-
-  /**
-   * Arithmetic operation - calculate LCM of two columns.
-   */
-  public static Integer lcm(Integer i1, Integer i2) {
-    if (i1 == null || i2 == null) {
-      return null;
+    switch (outputType) {
+      case SHORT:
+        short shortProd = 1;
+        for (Number num : nums) {
+          shortProd *= num.shortValue();
+        }
+        return shortProd;
+      case INTEGER:
+        int intProd = 1;
+        for (Number num : nums) {
+          intProd *= num.intValue();
+        }
+        return intProd;
+      case LONG:
+        long longProd = 1;
+        for (Number num : nums) {
+          longProd *= num.longValue();
+        }
+        return longProd;
+      case FLOAT:
+        float floatProd = 1;
+        for (Number num : nums) {
+          floatProd *= num.floatValue();
+        }
+        return floatProd;
+      case DOUBLE:
+        double doubleProd = 1;
+        for (Number num : nums) {
+          doubleProd *= num.doubleValue();
+        }
+        return doubleProd;
+      default:
+        BigDecimal bdProd = BigDecimal.valueOf(1);
+        for (Number num : nums) {
+          bdProd = bdProd.multiply(new BigDecimal(num.toString()));
```

Comment:
Same comment as before. No need to convert to BigDecimal if the number is already a BigDecimal.